src/utils/internal/optionalRedirect.ts

```diff
@@ -1,9 +1,10 @@
 import { safeRedirect } from '../redirect';
 
 export function optionalRedirect(
-  callbackUrl?: string,
-  shouldRedirect?: boolean
+  callbackUrl?: string
 ) {
+  const shouldRedirect = Boolean(callbackUrl);
```

Comment:
I think we could drop this guy and move the boolean(callbackUrl) in the if below

Reply:
If I do this, the below code complains about missing callbackUrl and I have to pass the default value in 2 places. That's why I kept the validation for callbackUrl in if statement

![image](https://user-images.githubusercontent.com/11082863/177151469-1b1943c8-f348-4d16-9e85-bf07596e3b98.png)

Reply:
I will come with a small change here which is more elegant approach

Reply:
@StanislavSava,  what about this?

```
export function optionalRedirect(
  callbackUrl?: string
) {
  if (Boolean(callbackUrl)) {
    const callbackRoute = callbackUrl ?? window.location.href;

    setTimeout(() => {
      if (!window.location.pathname.includes(callbackRoute)) {
        safeRedirect(callbackRoute);
      }
    }, 200);
  }
}
```